diff -Naur src/boost_1_65_0/boost/numeric/ublas/matrix.hpp new/boost_1_65_0/boost/numeric/ublas/matrix.hpp
--- src/boost_1_65_0/boost/numeric/ublas/matrix.hpp	2019-06-22 19:36:35.000000000 -0700
+++ new/boost_1_65_0/boost/numeric/ublas/matrix.hpp	2019-06-22 19:36:12.000000000 -0700
@@ -1384,7 +1384,7 @@
 
         /*! @note "pass by value" the key idea to enable move semantics */
         BOOST_UBLAS_INLINE
-        fixed_matrix &operator = (matrix m) {
+        fixed_matrix &operator = (fixed_matrix m) {
             assign_temporary(m);
             return *this;
         }
diff -Naur src/boost_1_65_0/tools/build/src/tools/builtin.jam new/boost_1_65_0/tools/build/src/tools/builtin.jam
--- src/boost_1_65_0/tools/build/src/tools/builtin.jam	2017-08-19 09:49:50.000000000 -0700
+++ new/boost_1_65_0/tools/build/src/tools/builtin.jam	2019-06-22 19:31:33.000000000 -0700
@@ -306,7 +306,7 @@
 
     # Advanced RISC Machines
     armv2 armv2a armv3 armv3m armv4 armv4t armv5 armv5t armv5te armv6 armv6j iwmmxt ep9312
-	armv7 armv7s
+	armv7 armv7s arm64
 
     : propagated optional ;
 
diff -Naur src/boost_1_65_0/tools/build/src/tools/darwin.jam new/boost_1_65_0/tools/build/src/tools/darwin.jam
--- src/boost_1_65_0/tools/build/src/tools/darwin.jam	2017-08-19 09:49:50.000000000 -0700
+++ new/boost_1_65_0/tools/build/src/tools/darwin.jam	2019-06-22 19:33:34.000000000 -0700
@@ -476,10 +476,10 @@
         {
             if $(address-model) = 32_64 {
                 options = -arch i386 -arch x86_64 ;
-            } else if $(address-model) = 64 {
-                options = -arch x86_64 ;
-            } else {
+            } else if $(address-model) = 32 {
                 options = -arch i386 ;
+            } else {
+                options = -arch x86_64 ;
             }
         }        
         
@@ -505,7 +505,7 @@
             if $(instruction-set) {
                 options = -arch$(_)$(instruction-set) ;
             } else {
-                options = -arch arm ;
+                options = -arch arm64 ;
             }
         }
     }
